app/controllers/katello/api/v2/content_uploads_controller.rb

```diff
@@ -13,6 +13,8 @@ class Api::V2::ContentUploadsController < Api::V2::ApiController
     param :content_type, RepositoryTypeManager.uploadable_content_types(false).map(&:label), :required => false, :desc => N_("content type ('deb', 'docker_manifest', 'file', 'ostree', 'rpm', 'srpm')")
     def create
       content_type = params[:content_type] || ::Katello::RepositoryTypeManager.find(@repository.content_type).default_managed_content_type.label
+      fail _("Content type is not supported for upload!") unless content_type
```

Comment:
I can cause two different errors here.

The first one is if I don't specify `content_type` in the API request, the return message is `"undefined method label for nil:NilClass","errors":["undefined method label for nil:NilClass"]`.

The second one is if I pass `ansible collections` as `content_type`, the return message is `"uninitialized constant Katello::Pulp3::AnsibleCollection::CONTENT_TYPE","errors":["uninitialized constant Katello::Pulp3::AnsibleCollection::CONTENT_TYPE"]`

But I think the solution you used for `repositories_controller` would work fine here.

Reply:
Can you check these on the updated commit..

I am a little surprised you were able to pass ansible_collections as a param and not fail at the param validation itself or maybe I am reading it wrong:
`param :content_type, RepositoryTypeManager.uploadable_content_types(false).map(&:label), :required => false, :desc => N_("content type ('deb', 'docker_manifest', 'file', 'ostree', 'rpm', 'srpm')")`

RepositoryTypeManager.uploadable_content_types(false).map(&:label) does not return ansible_collections so it should ideally fail before going any further. In any case, I added that check inside the controller to prevent that.